web/components/contract-search.tsx

```diff
@@ -207,24 +253,34 @@ function ContractSearchControls(props: {
     additionalFilter,
     hideOrderSelector,
     onSearchParametersChanged,
-    useQuerySortLocalStorage,
+    persistPrefix,
     useQuerySortUrlParams,
     user,
   } = props
 
-  const savedSort = useQuerySortLocalStorage ? getSavedSort() : null
-  const initialSort = savedSort ?? defaultSort ?? 'score'
-  const querySortOpts = { useUrl: !!useQuerySortUrlParams }
-  const [sort, setSort] = useSort(initialSort, querySortOpts)
-  const [query, setQuery] = useQuery('', querySortOpts)
-  const [filter, setFilter] = useState<filter>(defaultFilter ?? 'open')
-  const [pillFilter, setPillFilter] = useState<string | undefined>(undefined)
+  const localStore = safeLocalStorage()
+  const sessionStore = safeSessionStorage()
+  const persistAs = (name: string, store?: Storage) => {
+    return persistPrefix ? { prefix: persistPrefix, name, store } : undefined
+  }
 
-  useEffect(() => {
-    if (useQuerySortLocalStorage) {
-      setSavedSort(sort)
-    }
-  }, [sort])
+  const initialSort = defaultSort ?? 'score'
+  const [sort, setSort] = useSort(initialSort, {
+    useUrl: !!useQuerySortUrlParams,
+    persist: persistAs('sort', localStore),
+  })
+  const [query, setQuery] = useQuery('', {
+    useUrl: !!useQuerySortUrlParams,
+    persist: persistAs('query', sessionStore),
+  })
```

Comment:
Why do we need to persist the `query` in session storage? Wouldn't it be in the url?